File: client/src/types/survey.ts
export type SurveyStatus = 'draft' | 'published'

export type QuestionSingle = {
  id: string
  type: 'single'
  text: string
  options: string[]
}

export type QuestionText = {
  id: string
  type: 'text'
  text: string
}

export type Question = QuestionSingle | QuestionText

export type Survey = {
  id: string
  title: string
  description: string
  status: SurveyStatus
  questions: Question[]
}

export type SurveyResponseRow = {
  id: string
  submittedAt: string
  answers: Record<string, string>
}

export type AnswersMap = Record<string, string>

File: backend/app/models/survey.py
from sqlalchemy import Column, String, Text, DateTime, ForeignKey, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
import uuid
from app.core.database import Base


class Survey(Base):
    __tablename__ = "surveys"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    public_id = Column(String(50), unique=True, nullable=True)
    author_id = Column(UUID(as_uuid=True), ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
    title = Column(String(255), nullable=False)
    description = Column(Text, nullable=True)
    status = Column(String(20), default="draft")
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
    published_at = Column(DateTime(timezone=True), nullable=True)

    # Relationships
    author = relationship("User", back_populates="surveys")
    questions = relationship("Question", back_populates="survey", cascade="all, delete-orphan", order_by="Question.order_index")
    answers = relationship("Answer", back_populates="survey", cascade="all, delete-orphan")